src/main/java/com/thevoxelbox/voxelsniper/brush/DrainBrush.java

```diff
@@ -33,47 +34,37 @@ private void drain(final SnipeData v) {
 
-                for (int y = brushSize; y >= 0; y--) {
-                    if ((xSquared + Math.pow(y, 2)) <= brushSizeSquared) {
-                        if (this.getBlockMaterialAt(this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() + y) == Material.WATER || this.getBlockMaterialAt(this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() + y) == Material.LAVA) {
-                            undo.put(this.clampY(this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() + y));
-                            this.setBlockMaterialAt(this.getTargetBlock().getZ() + y, this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), Material.AIR);
-                        }
-
-                        if (this.getBlockMaterialAt(this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() - y) == Material.WATER || this.getBlockMaterialAt(this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() - y) == Material.LAVA) {
-                            undo.put(this.clampY(this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() - y));
-                            this.setBlockMaterialAt(this.getTargetBlock().getZ() - y, this.getTargetBlock().getX() + x, this.getTargetBlock().getY(), Material.AIR);
-                        }
-
-                        if (this.getBlockMaterialAt(this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() + y) == Material.WATER || this.getBlockMaterialAt(this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() + y) == Material.LAVA) {
-                            undo.put(this.clampY(this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() + y));
-                            this.setBlockMaterialAt(this.getTargetBlock().getZ() + y, this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), Material.AIR);
-                        }
-
-                        if (this.getBlockMaterialAt(this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() - y) == Material.WATER || this.getBlockMaterialAt(this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() - y) == Material.LAVA) {
-                            undo.put(this.clampY(this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), this.getTargetBlock().getZ() - y));
-                            this.setBlockMaterialAt(this.getTargetBlock().getZ() - y, this.getTargetBlock().getX() - x, this.getTargetBlock().getY(), Material.AIR);
+                for (int z = brushSize; z >= 0; z--) {
+                    if ((xSquared + Math.pow(z, 2)) <= brushSizeSquared) {
+                        for (int dx : new int[]{-1, 1}) {
+                            for (int dz : new int[]{-1, 1}) {
```

Comment:
This is an interesting way to do this, haven't seen it like this before

Reply:
If you know of a better way, let me know, but I found this better then (i=-1;i<=1;i+=2).
This prevents writing the same thing 4 times.

Reply:
Oh no this was a compliment, I am going to remember this idea for when I could use it